packages/gatsby/src/utils/js-chunk-names.ts

```diff
@@ -38,12 +38,33 @@ function replaceUnifiedRoutesKeys(
 
 const chunkNameCache = new Map()
 export function generateComponentChunkName(componentPath: string): string {
+  if (componentPath.indexOf(`?__mdxPath=`) !== -1) {
+    componentPath = componentPath.split(`?__mdxPath=`)[1]
+  }
   if (chunkNameCache.has(componentPath)) {
     return chunkNameCache.get(componentPath)
   } else {
     const { program } = store.getState()
     const directory = program?.directory || `/`
-    const name = pathRelative(directory, componentPath)
+    let name = pathRelative(directory, componentPath)
+
+    /**
+     * File names should not exceed 255 characters
+     * * minus 12 for `component---`
+     * * minus 7 for `.js.map`
+     */
+    const maxLength = 236
+
+    // Shorten path name to not exceed max length of 255 characters
+    if (name.length > maxLength) {
+      const hash = require(`crypto`)
+        .createHash(`sha1`)
+        .update(name)
+        .digest(`base64`)
+
+      // Shorten the name, ensure that adding the hash does not exceed 255 chars
+      name = `${name.substring(0, maxLength - 41)}-${hash}`
```

Comment:
How often does this happen? (go for md5 it's smaller and faster and go with hex?)

We can also use basename + hash instead of doing the full path

Reply:
just basename -> conflicts on `foo/index.mdx` + `bar/index.mdx`

Reply:
It happens in every case where an author/editor decides to use the full possible file name length in their OS. While I personally find it insane to use the whole title as file name, some might do.

Like here: https://github.com/wesbos/wesbos/blob/master/src/posts/Matt%20Mullenweg%20to%20WordPress%20Community%20-%20%20Learn%20JavaScript%20Deeply%20Here%20are%20some%20resources%20for%20doing%20that/Matt%20Mullenweg%20to%20WordPress%20Community%20-%20%20Learn%20JavaScript%20Deeply%20Here%20are%20some%20resources%20for%20doing%20that.mdx

Reply:
Interesting but we also do relative path so it's probably not going to happen that often. So this is fine :)